### PR TITLE
Issue 476 - Fix to check if custom attributes were changed from default value

### PR DESCRIPTION
**Pull request by rodfersou**

### CODE REVIEW

**Comment by rodfersou** [comment]
@hvelarde the tests are passing locally

**Comment by hvelarde** [comment]
@fredvd could you please check if this fix your problem?

File: CHANGES.rst [hunk 1]
```diff
@@ -25,6 +25,9 @@ There's a frood who really knows where his towel is.
 - Allow to choose a custom title and description on items in a carousel tile (closes `#459`_).
   [rodfersou]
 
+- Fix to check if custom attributes were changed from default value (closes `#476`_).
```
Comment (hvelarde) [inline CHANGES.rst]: changelog entries on top of the old ones, please

**Comment by hvelarde** [comment]
I know, but they must pass also remotely; what's the purpose of CI if not ;-)

**Comment by fredvd** [comment]
@hvelarde @rodfersou  Coincedentally this  pull request seems to fix issue #475 . I cannot reproduce 
that one. 

Concerning #476 for which this pull request was made: it fixes it partially, as long as there is plain ASCII in all the fields. Custom fields are still overwritten on the first save when the text contains unicode, so there is I guess something missing to handle unicode on the attribute change check. Here's an example uuids dict where I changed an O into an ö to the tekst on the Link Item that I dropped on the carousel:  Notice the 'xf6' in the unicode string. 


{'count': None, 'description': None, 'title': None, 'css_class': None, 'image': None, 'more_link_text': None, 'more_link': None, 'uuids': {u'a51e533497e149938b93984763a57f61': {u'order': u'0'}, u'bc2b473a5c85406d8ab10b40d5723bf4': {u'custom_description': u'\xf6penbare besturen banen al jarenlang de weg naar pesticidenvrij beheer. Vanaf januari gaan we een stap verder en moeten ook scholen, ziekenhuizen, kinderdagverblijven. pesticidenvrij beheerd worden.', u'order': u'1'}}, 'date': None, 'autoplay': True, 'tile_title': u'Custom Title'}

**Comment by frapell** [comment]
I haven't read the code, but for reading the title of the pull, I wonder... does this handle a case where the user customizes values for added elements to the carousel, and then decides to "go back"? 
In my code, I assumed if field was blank, then I should use the object's data, else, if had any value, use that... Is a case like this considered in this code?

**Comment by rodfersou** [comment]
@frapell I know that your code assumed that the field was blank.. but this is not the default behaviour of other tiles, so I think this is the right way to go.

**Comment by hvelarde** [comment]
@rodfersou please fix the tests so we can merge this

**Comment by hvelarde** [comment]
@fredvd can you please confirm that the issue is fixed?

**Pull request merged**